content/en/docs/concepts/services-networking/endpoint-slices.md

```diff
@@ -135,30 +134,25 @@ For pods, this is any pod that has a deletion timestamp set.
 
 ### Topology information {#topology}
 
-{{< feature-state for_k8s_version="v1.20" state="deprecated" >}}
+Each endpoint within an EndpointSlice can contain relevant topology information.
+The topology information includes the location of the endpoint and information
+about the corresponding Node and zone. This will be available in the following
+per endpoint fields on EndpointSlices:
+
+* `nodeName` - The name of the Node this endpoint is on.
+* `zone` - The zone this endpoint is in.
 
 {{< note >}}
-The topology field in EndpointSlices has been deprecated and will be removed in
-a future release. A new `nodeName` field will be used instead of setting
-`kubernetes.io/hostname` in topology. It was determined that other topology
-fields covering zone and region would be better represented as EndpointSlice
-labels that would apply to all endpoints within the EndpointSlice.
+In the GA API, the per endpoint `topology` was effectively removed in favor of
+the dedicated fields `nodeName` and `zone`.
+
+If the only keys used in the `topology` field were
+`kubernetes.io/hostname` and `topology.kubernetes.io/zone`, those values
+will be available in the per endpoint `nodeName` and `zone` fields
+respectively. Any other keys will require continued usage of the
+`v1beta1` API.
```

Comment:
This is a confusing thing to document, and I'm not sure how much detail is appropriate here. Maybe something along the lines of this would work:

> Setting arbitrary topology fields per endpoint has been deprecated and will not be supported in the v1 API. Instead, the v1 API supports setting individual node name and zone fields. These fields will be automatically translated between API versions. For example, the value of `topology[topology.kubernetes.io/zone]` in the v1beta1 API will be accessible in the `zone` field in the v1 API.

Reply:
That sounds much clearer, thanks!